### PR TITLE
EPMRPP-37204. React. Apply new preferences component

### PR DESCRIPTION
**Pull request by AmsterGet**

### CODE REVIEW

File: app/src/pages/inside/filtersPage/filterGrid/displayFilter/displayFilter.jsx [hunk 1]
```diff
@@ -7,31 +7,37 @@ import styles from './displayFilter.scss';
...
-            />
-          )}
-        </span>
-      </InputSwitcher>
-    </div>
-    <div className={cx('separator')} />
-  </Fragment>
-);
+export const DisplayFilter = ({ userFilters, filter, onChangeDisplay }) => {
+  const isFilterDisplayed = userFilters.find((item) => item.id === filter.id);
```
Comment (DiscoElevator) [inline app/src/pages/inside/filtersPage/filterGrid/displayFilter/displayFilter.jsx]: `const isFilterDisplayed = !!userFilters.find((item) => item.id === filter.id);`
`is<something>` usually means that it's a boolean

File: app/src/controllers/project/actionCreators.js [hunk 2]
```diff
@@ -44,17 +54,18 @@ export const updateProjectEmailConfig = (emailConfig) => (dispatch, getState) =>
   });
 };
 
-export const toggleDisplayFilterOnLaunchesAction = (filter) => (dispatch, getState) => {
-  dispatch({
-    type: TOGGLE_DISPLAY_FILTER_ON_LAUNCHES,
-    payload: filter,
-  });
-  dispatch(updateProjectPreferencesAction(projectPreferencesSelector(getState())));
+export const toggleDisplayFilterOnLaunchesAction = (filter, method, stateUpdate = true) => (
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/controllers/project/actionCreators.js]: It indicates whether to update the redux state or not.

[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/controllers/project/actionCreators.js]: @DiscoElevator Do you want to suggest writing separate actions for `add` and `delete`?

File: app/src/controllers/project/actionCreators.js [hunk 3]
```diff
@@ -1,35 +1,45 @@
...
   type: UPDATE_AUTO_ANALYSIS_CONFIGURATION,
   payload: project.configuration.attributes,
 });
 
-const updateProjectPreferencesAction = (settings) => (dispatch, getState) =>
-  fetch(URLS.projectPreferences(activeProjectSelector(getState()), userIdSelector(getState())), {
-    method: 'PUT',
-    data: settings,
-  });
+const updateProjectPreferencesAction = (filterId, method) => (dispatch, getState) =>
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/controllers/project/actionCreators.js]: Changed to `updateProjectFilterPreferencesAction`

File: app/src/controllers/project/reducer.test.js [hunk 4]
```diff
@@ -86,25 +86,65 @@ describe('project reducer', () => {
...
-      const stateWithFilter = projectPreferencesReducer(oldState, {
-        type: TOGGLE_DISPLAY_FILTER_ON_LAUNCHES,
+      const payload = { id: 'filter1' };
+      const stateWithoutFilter = projectPreferencesReducer(oldState, {
+        type: UPDATE_FILTER_SUCCESS,
         payload,
       });
+      expect(stateWithoutFilter).toEqual({
+        ...oldState,
+        filters: [...oldState.filters, payload],
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/controllers/project/reducer.test.js]: Fixed

File: app/src/pages/inside/filtersPage/filterGrid/displayFilter/displayFilter.jsx (hunk 1 above)
Comment (AmsterGet) [reply app/src/pages/inside/filtersPage/filterGrid/displayFilter/displayFilter.jsx]: Sure. Fixed

File: app/src/controllers/project/actionCreators.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/controllers/project/actionCreators.js]: We decided to create separated actions for `add` and `delete`.

**Pull request merged**